src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -622,21 +621,17 @@ public boolean shouldExpire()
             return true;
         }
 
-        PacketIOActivity packetIOActivity
-                = this.transceiver.getPacketIOActivity();
-
-        int maxExpireTimeSecsFromChannelShims = channelShims.stream()
+        Duration maxExpireTimeFromChannelShims = channelShims.stream()
                 .map(ChannelShim::getExpire)
-                .mapToInt(exp -> exp)
-                .max()
-                .orElse(0);
-
-        long lastActivity
-                = packetIOActivity.getLastOverallActivityTimestampMs();
-        long now = System.currentTimeMillis();
-        if (lastActivity <= 0)
+                .map(Duration::ofSeconds)
+                .max(Comparator.comparing(Function.identity()))
+                .orElse(Duration.ofSeconds(0));
+
+        Instant lastActivity = getLastActivity();
+        Instant now = clock.instant();
+        if (lastActivity == ClockUtils.NEVER)
         {
-            Duration timeSinceCreation = Duration.ofMillis(now - creationTimeMillis);
+            Duration timeSinceCreation = Duration.between(creationTimeMillis, now);
             if (timeSinceCreation.compareTo(EP_TIMEOUT) > 0) {
                 logger.info("Endpoint's ICE connection has neither failed nor connected " +
                     "after " + timeSinceCreation + ", expiring");
```

Comment:
Now we're getting to the point where we might want a more human-friendly equivalent of `Duration.toString()`, but that can be for a separate patch.

Reply:
sounds good